File: android/src/main/kotlin/com/usercentrics/sdk/flutter/bridge/TrackBridge.kt
package com.usercentrics.sdk.flutter.bridge

import com.usercentrics.sdk.UsercentricsAnalyticsEventType
import com.usercentrics.sdk.flutter.api.FlutterMethodCall
import com.usercentrics.sdk.flutter.api.FlutterResult
import com.usercentrics.sdk.flutter.api.UsercentricsProxy
import com.usercentrics.sdk.flutter.api.UsercentricsProxySingleton

internal class TrackBridge(
    private val usercentrics: UsercentricsProxy = UsercentricsProxySingleton
) : MethodBridge {

    override val name: String
        get() = "track"

    override fun invoke(call: FlutterMethodCall, result: FlutterResult) {
        assert(name == call.method)

        val argsMap = call.arguments as Map<*, *>
        usercentrics.instance.track(UsercentricsAnalyticsEventType.valueOf(argsMap["event"] as String))

        result.success(null)
    }
}

File: ios/Classes/Bridge/TrackBridge.swift
import Usercentrics

struct TrackBridge : MethodBridge {

    let name: String = "track"
    let usercentrics: UsercentricsProxyProtocol

    func invoke(_ call: FlutterMethodCall, _ result: @escaping FlutterResult) {
        assert(call.method == name)

        let argsDict = call.arguments as! Dictionary<String, Any>
        let event = UsercentricsAnalyticsEventType.initialize(from: argsDict["event"])!
        usercentrics.shared.track(event: event)

        result(nil)
    }
}
